frontend/src/stimulus/controllers/dynamic/admin/custom-fields.controller.ts

```diff
@@ -236,15 +242,12 @@ export default class CustomFieldsController extends Controller {
   }
 
   private toggleFormat(format:string) {
-    this.setActive(this.allowNonOpenVersionsTargets, format === 'version');
-    this.setActive(this.defaultBoolTargets, format === 'bool');
-    this.setActive(this.defaultLongTextTargets, format === 'text');
-    this.setActive(this.defaultTextTargets, !['list', 'bool', 'date', 'text', 'user', 'version'].includes(format));
-    this.setActive(this.lengthTargets, !['list', 'bool', 'date', 'user', 'version', 'link'].includes(format));
-    this.setActive(this.multiSelectTargets, ['list', 'user', 'version'].includes(format));
-    this.setActive(this.possibleValuesTargets, format === 'list');
-    this.setActive(this.regexpTargets, !['list', 'bool', 'date', 'user', 'version'].includes(format));
-    this.setActive(this.searchableTargets, !['bool', 'date', 'float', 'int', 'user', 'version'].includes(format));
-    this.setActive(this.textOrientationTargets, format === 'text');
+    this.formatConfigValue.forEach(([targetsName, operator, formats]) => {
+      const active = operator == 'only' ? formats.includes(format) : !formats.includes(format);
+      const targets = this[`${targetsName}Targets` as keyof typeof this] as HTMLElement[];
+      if (targets) {
+        this.setActive(targets, active);
+      }
+    });
```

Comment:
❓ admittedly I don't get how this get's rid of the blinking.

Reply:
Frontend part doesn't really change, just config is moved to backend, so it is backend part that gets rid of blinking, as fields are at start hidden or visible depending on the field format, so before stimulus toggles them